File: ExtremeStudio/ExtremeParser/Types/VarStruct.vb
Public Structure VarStruct

    Public VarName As String
    Public VarTag As String
    Public VarArrays As List(Of String)
    Public DefaultValue As String

    Public Sub New(name As String, tag As String, def As String, arrays As List(Of String))
        VarName = name
        VarTag = tag
        VarArrays = arrays
        DefaultValue = def
    End Sub

    Public Shared Operator =(ByVal first As VarStruct, second As VarStruct)
        If first.VarName = second.VarName And first.VarTag = second.VarTag And first.DefaultValue = second.DefaultValue Then
            Return True
        End If
        Return False
    End Operator
    Public Shared Operator <>(ByVal first As VarStruct, second As VarStruct)
        If first = second Then
            Return False
        Else
            Return True
        End If
    End Operator

End Structure

File: ExtremeStudio.Parser/Types/VarStruct.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExtremeStudio.Parser.Types
{
    public struct VarStruct
    {

        public string VarName;
        public string VarTag;
        public List<string> VarArrays;
        public string DefaultValue;

        public VarStruct(string name, string tag, string def, List<string> arrays)
        {
            VarName = name;
            VarTag = tag;
            VarArrays = arrays;
            DefaultValue = def;
        }

        public static object operator ==(VarStruct first, VarStruct second)
        {
            if (first.VarName == second.VarName && first.VarTag == second.VarTag && first.DefaultValue == second.DefaultValue)
            {
                return true;
            }
            return false;
        }
        public static object operator !=(VarStruct first, VarStruct second)
        {
            if ((bool)(first == second))
            {
                return false;
            }
            else
            {
                return true;
            }
        }

        public override bool Equals(object o)
        {
            throw new NotImplementedException();
        }
        public override int GetHashCode()
        {
            throw new NotImplementedException();
        }

    }
}
